src/plugins/mpi/pmix/mpi_pmix.c

```diff
@@ -240,5 +313,34 @@ extern mpi_plugin_client_state_t *p_mpi_hook_client_prelaunch(
 
 extern int p_mpi_hook_client_fini(void)
 {
-	return SLURM_SUCCESS;
+	// recv abort code
```

Comment:
We don't need this:
* stepd doesn't need to return error code (it's err code is not propagated anywhere)
* srun has this value in it's Virtual Address Space.

Reply:
Use pmix_info database to store error code on the srun side.

Reply:
Need to finalize abort thread here!
implement through `pmixp_abort_agent_stop()` (locate in pmixp_server.c)